src/lib.rs

```diff
@@ -388,19 +410,12 @@ where
     /// * This function can return an `Utf8Error` if there was an error with the generation of the
     /// accept string. This should also be impossible but an error is preferable to a panic
     /// * Returns `WebsocketAlreadyOpen` if called on a websocket that is already open
-    ///
-    /// # Panics
-    /// Panics with `Server websocket expected` if the user calls this function as a client
-    /// websocket. This function should be use with a websocket created using `new_server()`
     pub fn server_accept(
         &mut self,
         sec_websocket_key: &WebSocketKey,
         sec_websocket_protocol: Option<&WebSocketSubProtocol>,
         to: &mut [u8],
     ) -> Result<usize> {
-        if self.is_client {
```

Comment:
Really great that you got rid of the need for this. There is still a panic in the client_accept function that can now be removed too.